### PR TITLE
test(state) remove and test filter undefined

### LINKED ISSUE #151: @rx-angular/state. Undefined values from observable source are filtered out in connect method.
Hi.

Approach with filtering out undefined values totally makes sense in select method but in connect situation is a bit different. It is a pretty common case to create a bunch of subjects that have no value and serves only as an event trigger.

Example

```
// TEMPLATE
// not working
<button (click)="insertCreature$.next()">Insert</button>  
// working
<button (click)="insertCreature$.next(null)">Insert</button>

// CLASS
readonly insertCreature$ = new Subject<void>();
  constructor() {
    super();
    this.connect("creatures", this.insertCreature$, ({ creatures }) => {
      console.log('will not fire');
      const creatureToAdd = { id: 42, name: "newCreature", type: "dinosaur" };
      return [...(creatures || []), creatureToAdd];
    });
  }
```

stackblitz: https://stackblitz.com/edit/empty-subject-value-bug?file=src%2Fapp%2Fapp.component.ts

Suggestion: remove filtering logic in the connect method if there are no downsides.

### PR DESCRIPTION
**Pull request by BioPhoton**
closes #151

### CODE REVIEW

File: libs/state/src/lib/rx-state.service.ts [hunk 1]
```diff
@@ -244,55 +244,51 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
     projectOrSlices$?: ProjectStateReducer<T, V> | Observable<T[K]>,
     projectValueFn?: ProjectValueReducer<T, K, V>
   ): void {
+    // state.connect(observable$, undefined, undefined)
```
Comment (hoebbelsB) [inline libs/state/src/lib/rx-state.service.ts:247]: ???

File: libs/state/src/lib/rx-state.service.ts [hunk 2]
```diff
@@ -244,55 +244,51 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
...
     ) {
-      const slice$ = keyOrInputOrSlice$.pipe(
-        filter(slice => slice !== undefined)
-      );
-      this.accumulator.nextSliceObservable(slice$);
+      this.accumulator.nextSliceObservable(keyOrInputOrSlice$);
       return;
     }
 
+    // state.connect(observable$, s => s, undefined)
```
Comment (hoebbelsB) [inline libs/state/src/lib/rx-state.service.ts:257]: ??

File: libs/state/src/lib/rx-state.service.ts [hunk 3]
```diff
@@ -244,55 +244,51 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
...
-      const slice$ = keyOrInputOrSlice$.pipe(
-        filter(slice => slice !== undefined),
-        map(v => project(this.get(), v))
-      );
+      const slice$ = keyOrInputOrSlice$.pipe(map(v => project(this.get(), v)));
       this.accumulator.nextSliceObservable(slice$);
       return;
     }
 
+    // state.connect(observable$, updates$, undefined)
```
Comment (hoebbelsB) [inline libs/state/src/lib/rx-state.service.ts:270]: ???

File: libs/state/src/lib/rx-state.service.ts [hunk 4]
```diff
@@ -244,55 +244,51 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
...
       const key = keyOrInputOrSlice$;
       const slice$ = projectOrSlices$.pipe(
-        filter(slice => slice !== undefined),
         map(value => ({ ...{}, [key]: value }))
       );
       this.accumulator.nextSliceObservable(slice$);
       return;
     }
 
+    // state.connect(observable$, updates$, (s) => s.v))
```
Comment (hoebbelsB) [inline libs/state/src/lib/rx-state.service.ts:284]: ???

File: libs/state/src/lib/rxjs/interfaces/compare-fn.ts [hunk 5]
```diff
@@ -3,9 +3,6 @@
  * The function which is used by `KeyCompareMap` to determine if changes are distinct or not.
  * Should return true if values are equal.
  *
- * @param {T | null | undefined} oldVal
- * @param {T | null | undefined} newVal
- *
```
Comment (hoebbelsB) [inline libs/state/src/lib/rxjs/interfaces/compare-fn.ts:8]: i guess this went wrong?

File: libs/state/src/lib/rx-state.service.ts [hunk 6]
```diff
@@ -336,55 +336,51 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
     projectOrSlices$?: ProjectStateReducer<T, V> | Observable<T[K]>,
     projectValueFn?: ProjectValueReducer<T, K, V>
   ): void {
+    // state.connect(observable$, undefined, undefined)
```
Comment (BioPhoton) [inline libs/state/src/lib/rx-state.service.ts:339]:
```suggestion
```

File: libs/state/src/lib/rx-state.service.ts [hunk 7]
```diff
@@ -341,50 +341,45 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
...
     ) {
-      const slice$ = keyOrInputOrSlice$.pipe(
-        filter(slice => slice !== undefined)
-      );
-      this.accumulator.nextSliceObservable(slice$);
+      this.accumulator.nextSliceObservable(keyOrInputOrSlice$);
       return;
     }
 
+    // state.connect(observable$, s => s, undefined)
```
Comment (BioPhoton) [inline libs/state/src/lib/rx-state.service.ts:348]:
```suggestion
```

File: libs/state/src/lib/rx-state.service.ts [hunk 8]
```diff
@@ -355,36 +352,33 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
...
-      const slice$ = keyOrInputOrSlice$.pipe(
-        filter(slice => slice !== undefined),
-        map(v => project(this.get(), v))
-      );
+      const slice$ = keyOrInputOrSlice$.pipe(map(v => project(this.get(), v)));
       this.accumulator.nextSliceObservable(slice$);
       return;
     }
 
+    // state.connect(observable$, updates$, undefined)
```
Comment (BioPhoton) [inline libs/state/src/lib/rx-state.service.ts:360]:
```suggestion
```

File: libs/state/src/lib/rx-state.service.ts [hunk 9]
```diff
@@ -370,21 +364,20 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
...
       const key = keyOrInputOrSlice$;
       const slice$ = projectOrSlices$.pipe(
-        filter(slice => slice !== undefined),
         map(value => ({ ...{}, [key]: value }))
       );
       this.accumulator.nextSliceObservable(slice$);
       return;
     }
 
+    // state.connect(observable$, updates$, (s) => s.v))
```
Comment (BioPhoton) [inline libs/state/src/lib/rx-state.service.ts:373]:
```suggestion
```

**Pull request merged**